codice/ddf
# [WIP] - DO NOT REVIEW - DDF-4133 Solr basic auth

#### What does this PR do?
This PR changes the method of authenticating to Solr from 2-way TLS to 1-way TLS with basic authorization.

- Changes Solr configuration to use basic authentication 
- Changes Solr configuration to use 1-way TLS
- Creates a solr user named "ddf" with passsword "changeit"
- Adds system properties send basic auth information in Solr HTTP requests
- Creates a new API and Impl, and associated bean, service, and metatype so that the password can be set from the Admin UI. The config admin will encrypt the password at rest.
- Adds documentation instructing users to set a strong password for Solr 

#### Who is reviewing it? 
@stustison @paouelle @mcalcote @garrettfreibott @ricklarsen @austinsteffes 

#### Select relevant component teams: 
@codice/security 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@stustison

#### How should this be tested?
- Build and install DDF
- Test different catalog and workspace operations to ensure all Solr core are working correctly
- Follow the hardening instructions to change the password on both unix systems and Windows.

#### What are the relevant tickets?
DDF-4133

#### Checklist:
- [ N/A] Update / Add Unit Tests
- [N/A] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
